### PR TITLE
test: Use SCA to register

### PR DESCRIPTION
**Pull request by m-horky**

### CODE REVIEW

**Comment by jirihnidek** [comment]
I would like to just mark failing test case skipped and resolve this issue later. I want to move forward.

**Comment by ptoscano** [comment]
> I would like to just mark failing test case skipped and resolve this issue later. I want to move forward.

That isn't exactly a good solution, especially when there only few test cases. Skipping any means effectively starting to not test things anymore.

I'll look into the candlepin test data, and see what can be done.

**Comment by ptoscano** [comment]
> I'll look into the candlepin test data, and see what can be done.

This was done few days ago, and at least it seemed working for me. The "services" cockpit image that contains the Candlepin container was just refreshed, so hopefully it should be fine here too.

Would you please rebase this?

File: test/check-subscriptions [hunk 1]
```diff
@@ -31,25 +31,29 @@ import testvm
 
 # candlepin on the services image has a lot of demo data preloaded
 # useful info/commands:
-#    Login: doc      password: password
-#    org:   snowwhite
+#    login:          huey
+#    password:       password
+#    organizations:  admin, snowwhite, donallduck
```
Comment (ptoscano) [inline test/check-subscriptions:36]: typo s/donallduck/donaldduck/

**Comment by ptoscano** [comment]
The activation key script (see `ACTIVATION_KEY_SCRIPT`) needs to be adapted to the different user: see `e["owner"]["displayName"]` and `e["owner"]["key"]`.

Since SCA now enables automatically all the content of the organization, this means `redhat.repo` gets enabled repositories by default; configured repositories have
```
gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-candlepin
```
which is not available in the system, and needs to be downloaded from Candlepin; furthermore, the `candlepin` container in the `services` image does not expose the port 8080 for content, so also dnf is not able to download the metadata of repositories. At least those issues causes the failure of `TestSubscriptionsPackages.testMissingPackages`. I'll fix the bits in the `services` image of cockpit.

Not sure about the rest of the failures, will need to dig more.

File: test/check-subscriptions [hunk 2]
```diff
@@ -244,7 +248,7 @@ class TestSubscriptions(SubscriptionsCase):
         b.wait_not_in_text("body", "Invalid credentials")
 
         # wait for message that we need to specify our org
-        b.wait_in_text("body", "User doc is member of more organizations, but no organization was selected")
+        b.wait_in_text("body", "User huey is member of more organizations, but no organization was selected")
 
         # now specify the org
         b.set_input_text("#subscription-register-org", "snowwhite")
```
Comment (ptoscano) [inline test/check-subscriptions:254]: this needs to be changed to "donaldduck"

**Comment by ptoscano** [comment]
After a bit more debugging and analysis, I know what happens and what to do.

------------

Regarding the tests `TestSubscriptions.testRegister` and `TestSubscriptions.testRegisterWithKey`: they try to verify that, after the registration, certain products are registered (checking the "Registered" green item next to each product). The problem is that, in SCA mode, the concept of "products" is very reduced, and there are no such info on the status of products; this means that those checks do not make sense anymore. Hence, considering the account used for testing is switched to an SCA one, all the `wait_subscription()` bits can be simply dropped. Please do that, mentioning the reasons in the commit message.

------------

Regarding the `TestSubscriptionsPackages.testMissingPackages` test: I mentioned in my previous comment the reason, i.e. we need to set up the repository management in subscription-manager properly:
1. copy the RPM GPG key from Candlepin
2. set the baseurl in the configuration

Both are easy to do, and I have the changes for them locally. The problem is that the `candlepin` container in the `services` cockpit image does not expose the port 8080, and thus the repositories cannot be accessed. I created earlier https://github.com/cockpit-project/bots/pull/6708 for the cockpit CI, and it will require the refresh of the `services` image -- not a complicated process, though.

------------

The other few bits I mentioned inline needs to be fixed too :)

**Review by ptoscano — APPROVED** [review]
LGTM now, thanks!

**Pull request merged**